api/controller/controllerapi/controllerapi.go

```diff
@@ -0,0 +1,106 @@
+package controllerapi
```

Comment:
This doesn't belong in the api layer IMO. The api layer is the front end to client api network calls. What we have here is 2 fold:
- a worker to hook into the dependency engine
- backend / server side logic to interact with State

A bit hand wavy, but this would need to be split in two - 1. a business layer implementing the service oriented aspect of performing the api request, interacting with the data model / state etc; and 2. the glue which binds the service to the dependency engine. Ideally the service/business layer would be implemented totally oblivious to how such services would be hosted. It's sort of what we should have done with the apiserver facade stuff, except we interwove network marshalling with the service/business layer. So in other words, if the business logic we layered properly, it could be plugged into either the existing apiserver facade stuff to received networked api calls, or directly into the dependency engine to receive local api calls.